subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -309,7 +288,7 @@ static int tls_init(const struct device *unused)
 		return -EFAULT;
 	}
 
-#if defined(MBEDTLS_DEBUG_C) && (CONFIG_NET_SOCKETS_LOG_LEVEL >= LOG_LEVEL_DBG)
+#if defined(MBEDTLS_DEBUG_C) && (CONFIG_MBEDTLS_DEBUG)
```

Comment:
The `dbg_init()` and relevant setup calls should be called somewhere more generic place, one option is `kernel/init.c` just before `main()` is called.

Reply:
I'm afraid that won't work, `mbedtls_dbg_init()` and thus `mbedtls_ssl_conf_dbg()` requires a TLS context structure, and you get this only after socket creation.

Reply:
Another thing is, there already is a mbedTLS initialized, but it's placed in the `mbedtls` tree: https://github.com/zephyrproject-rtos/mbedtls/blob/master/zephyr_init.c. I wonder if this should rather be moved into the Zephyr tree since it's a glue code for Zephyr. We could configure the threshold in there then. But it's probably out of the scope of this PR.

Reply:
> I wonder if this should rather be moved into the Zephyr tree since it's a glue code for Zephyr. We could configure the threshold in there then. But it's probably out of the scope of this PR.

Sounds like a plan.

Reply:
@rlubos Please correct me if I'm wrong, but you suggest moving mbedtls_dbg.c file to modules/crypto/mbedtls/mbedtls_dbg.c and then keep the call in sockets_tls.c, and then move the setting of logging threshold to zephyr_init?

Just asking for clarification to reduce unnecessary wrong commits.

I'm aiming for this way since it seems cleaner and also since I kinda have a hard time understanding how CMake includes source files in zephyr/modules/* :smile:

Reply:
> Please correct me if I'm wrong, but you suggest moving mbedtls_dbg.c file to modules/crypto/mbedtls/mbedtls_dbg.c and then keep the call in sockets_tls.c, and then move the setting of logging threshold to zephyr_init?

Correct. As for the `zephyr_init`, I'd consider moving it into the Zephyr three from the mbedTLS module, but I understand if you don't want to touch it as it's a bit out of scope of this PR. In such case, I guess we could leave the threshold setting as it is for now.


> I'm aiming for this way since it seems cleaner and also since I kinda have a hard time understanding how CMake includes source files in zephyr/modules/*
hal_nordic already does it, you can use it as an example: https://github.com/zephyrproject-rtos/zephyr/tree/master/modules/hal_nordic. I guess you just need to add top-level CMakeLists.txt in the module top folder which will add the source files.